field/composite.go

```diff
@@ -558,9 +563,22 @@ func (f *Composite) unpackSubfieldsByTag(data []byte) (int, error) {
 		tag := string(tagBytes)
 		if _, ok := f.spec.Subfields[tag]; !ok {
 			if f.skipUnknownTLVTags() {
-				// Obtain the length of the unknown tag and add it to the offset.
-				// Because BER-TLV lengths are decoded dynamically, the maxLen method argument is ignored.
-				fieldLength, readed, err := prefix.BerTLV.DecodeLength(ignoredMaxLen, data[offset:])
+				var (
+					pref   prefix.Prefixer
+					maxLen int
+				)
+				// to obtain the length of the unknown tag and add it to the offset we need to decode its length
+				if f.spec.Tag.Enc != encoding.BerTLVTag {
```

Comment:
What if `f.spec.Tag.PrefUnknownTLV` is `nil`?

To keep it backward compatible, I would rely on the `if f.spec.Tag.PrefUnknownTLV != nil { ... } else { use as it was before } `

Reply:
It is not possible that `f.spec.Tag.Enc != encoding.BerTLVTag && f.spec.Tag.PrefUnknownTLV == nil`)  as we previously check with:
```go
func (f *Composite) skipUnknownTLVTags() bool {
	return f.spec.Tag != nil && f.spec.Tag.SkipUnknownTLVTags && (f.spec.Tag.Enc == encoding.BerTLVTag || f.spec.Tag.PrefUnknownTLV != nil)
}
```

Please find this update, is it suitable for you?